pkg/metrics/metrics.go

```diff
@@ -565,6 +568,80 @@ var (
 	// APILimiterProcessedRequests is the counter of the number of
 	// processed (successful and failed) requests
 	APILimiterProcessedRequests = NoOpCounterVec
+
+	// WorkQueueDepth is the depth of the workqueue
+	//
+	// We set actual metrics here instead of NoOp for the workqueue metrics
+	// because these metrics will be registered with workqueue.SetProvider
+	// by init function in watcher.go. Otherwise, we will register NoOps.
+	//
+	WorkQueueDepth = metric.NewGaugeVec(metric.GaugeOpts{
+		ConfigName: Namespace + "_" + SubsystemWorkQueue + "_depth",
```

Comment:
nit: Perhaps a helper here might be in order to dry out the config naming?

Reply:
Sorry, I'm not following. Can you elaborate?

Reply:
If you are taking about the order, the order here is the same as the order of fields in `MetricsProvidor`
https://github.com/cilium/cilium/blob/e6748209b27ed24b319265ac45b517b9af9b516f/vendor/k8s.io/client-go/util/workqueue/metrics.go#L173-L181